src/components/ToggleSwitch.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import styled from 'styled-components/macro'
+
+export default function ToggleSwitch({ toggled, onChange }) {
+  return (
+    <div>
+      <Switch id="test-switch" toggled={toggled} onChange={onChange} />
+    </div>
+  )
+}
+
+const Switch = ({ id, toggled, onChange }) => {
+  return (
+    <>
+      <SwitchContainer>
+        <SpendingRed>Spending</SpendingRed>
+        <SwitchInput
+          className="switch-checkbox"
+          id={id}
+          type="checkbox"
+          checked={toggled}
+          onChange={onChange}
+        />
+        <SwitchLabel toggled={toggled} className="switch-label" htmlFor={id}>
+          <SwitchButton className="switch-button"></SwitchButton>
+        </SwitchLabel>
+        <IncomeGreen>Income</IncomeGreen>
+      </SwitchContainer>
+    </>
+  )
+}
+
+const SwitchContainer = styled.div`
+  display: flex;
+  justify-content: center;
+`
+const SpendingRed = styled.div`
+  color: red;
+  font-weight: bold;
+`
+const IncomeGreen = styled.div`
+  color: green;
+  font-weight: bold;
+`
+const toggleColors = {
+  true: '#36ffa1', //entspricht income
+  false: '#ff7a78', //entspricht spending
+}
+
+const SwitchInput = styled.input`
```

Comment:
maybe you can name your component SwitchInput "HiddenSwitchInput" so that it points out it's hidden?